--- SystemUI/Android.mk
+++ SystemUI/Android.mk
@@ -4,6 +4,7 @@
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_SRC_FILES := $(call all-java-files-under, src) \
+    $(call all-java-files-under,../SystemUIExt/src) \
     src/com/android/systemui/EventLogTags.logtags
 
 LOCAL_STATIC_JAVA_LIBRARIES := Keyguard \
@@ -22,6 +23,7 @@
 
 LOCAL_RESOURCE_DIR := \
     frameworks/base/packages/Keyguard/res \
+    $(LOCAL_PATH)/../SystemUIExt/res \
     $(LOCAL_PATH)/res
 LOCAL_AAPT_FLAGS := --auto-add-overlay --extra-packages com.android.keyguard
 
