--- SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
+++ SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
@@ -187,6 +187,7 @@
 import com.android.systemui.statusbar.policy.KeyguardUserSwitcher;
 import com.android.systemui.statusbar.policy.LocationControllerImpl;
 import com.android.systemui.statusbar.policy.NetworkControllerImpl;
+import com.android.systemui.statusbar.policy.NetworkTraffic;
 import com.android.systemui.statusbar.policy.NextAlarmController;
 import com.android.systemui.statusbar.policy.PreviewInflater;
 import com.android.systemui.statusbar.policy.RotationLockControllerImpl;
@@ -378,6 +379,9 @@
 
     private boolean mShowCarrierInPanel = false;
 
+    // Status bar Network traffic;
+    private NetworkTraffic mNetworkTraffic;
+
     // position
     int[] mPositionTmp = new int[2];
     boolean mExpandedVisible;
@@ -1010,6 +1014,8 @@
         mMoreIcon = mStatusBarView.findViewById(R.id.moreIcon);
         mNotificationIcons.setOverflowIndicator(mMoreIcon);
         mStatusBarContents = (LinearLayout)mStatusBarView.findViewById(R.id.status_bar_contents);
+
+        mNetworkTraffic = (NetworkTraffic)mStatusBarView.findViewById(R.id.network_traffic);
 
         mClockView = (TextView) mStatusBarView.findViewById(R.id.clock);
         mClockLocation = Settings.System.getIntForUser(mContext.getContentResolver(),
@@ -3860,6 +3866,7 @@
         animateCollapsePanels();
         updatePublicMode(); 
         updateNotifications();
+        mNetworkTraffic.updateSettings();
         resetUserSetupObserver();
         setControllerUsers();
 
