--- SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
+++ SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
@@ -267,7 +267,9 @@
 
     private final void updateHeadset(Intent intent) {
         int state = intent.getIntExtra("state", 0);
-        mService.setIconVisibility(SLOT_HEADSET, state == 1 ? true : false);
+        boolean mHeadsetIcon = Settings.System.getInt(mContext.getContentResolver(),
+                Settings.System.SHOW_HEADSET_ICON, 1) == 1;
+        mService.setIconVisibility(SLOT_HEADSET, (state == 1 && mHeadsetIcon)  ? true : false);
     }
 
     private final void updateSDCardtoAbsent() {
