--- SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
+++ SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
@@ -83,6 +83,7 @@
     private static final String SLOT_VOLUME = "volume";
     private static final String SLOT_CDMA_ERI = "cdma_eri";
     private static final String SLOT_ALARM_CLOCK = "alarm_clock";
+    private static final String SLOT_HEADSET = "headset";
     private static final String SLOT_SU = "su";
 
     private static final String SDCARD_ABSENT = "sdcard_absent";
@@ -134,6 +135,9 @@
             }
             else if (action.equals(Intent.ACTION_USER_SWITCHED)) {
                 updateAlarm();
+            }
+            else if (action.equals(Intent.ACTION_HEADSET_PLUG)) {
+                updateHeadset(intent);
             }
         }
     };
@@ -163,6 +167,7 @@
         filter.addAction(TelephonyIntents.ACTION_SIM_STATE_CHANGED);
         filter.addAction(TelecomManager.ACTION_CURRENT_TTY_MODE_CHANGED);
         filter.addAction(Intent.ACTION_USER_SWITCHED);
+	filter.addAction(Intent.ACTION_HEADSET_PLUG);
         mContext.registerReceiver(mIntentReceiver, filter, null, mHandler);
 
         int numPhones = TelephonyManager.getDefault().getPhoneCount();
@@ -199,6 +204,10 @@
         mService.setIcon(SLOT_VOLUME, R.drawable.stat_sys_ringer_vibrate, 0, null);
         mService.setIconVisibility(SLOT_VOLUME, false);
         updateVolumeZen();
+
+        // headset
+        mService.setIcon(SLOT_HEADSET, R.drawable.stat_sys_headset, 0, null);
+        mService.setIconVisibility(SLOT_HEADSET, false);
 
         if (mContext.getResources().getBoolean(R.bool.config_showSdcardAbsentIndicator)) {
             mStorageManager = (StorageManager) context
@@ -255,6 +264,11 @@
             onChange(selfChange, null);
         }
     };
+
+    private final void updateHeadset(Intent intent) {
+        int state = intent.getIntExtra("state", 0);
+        mService.setIconVisibility(SLOT_HEADSET, state == 1 ? true : false);
+    }
 
     private final void updateSDCardtoAbsent() {
         mService.setIcon(SDCARD_ABSENT, R.drawable.stat_sys_no_sdcard, 0, null);
