--- core/java/android/provider/Settings.java
+++ core/java/android/provider/Settings.java
@@ -3554,6 +3554,12 @@
          * @hide
          */
         public static final String STATUSBAR_COMPONENT = "statusbar_component";
+
+	/**
+         * Switch for headset icon
+         * @hide
+         */
+        public static final String SHOW_HEADSET_ICON = "show_headset_icon";
 
         /**
          * Settings to backup. This is here so that it's in the same place as the settings
