--- SystemUI/src/com/android/systemui/statusbar/phone/QSTileHost.java
+++ SystemUI/src/com/android/systemui/statusbar/phone/QSTileHost.java
@@ -62,6 +62,7 @@
 import com.android.systemui.qs.tiles.SyncTile;
 import com.android.systemui.qs.tiles.UsbTetherTile;
 import com.android.systemui.qs.tiles.VisualizerTile;
+import com.android.systemui.qs.tiles.ScreenshotTile;
 import com.android.systemui.qs.tiles.ScreenTimeoutTile;
 import com.android.systemui.qs.tiles.WifiTile;
 import com.android.systemui.settings.CurrentUserTracker;
@@ -376,6 +377,8 @@
                 return new HeadsUpTile(this);
             case QSConstants.TILE_AMBIENT_DISPLAY:
                 return new AmbientDisplayTile(this);
+            case QSConstants.TILE_SCREENSHOT:
+                return new ScreenshotTile(this);
             case QSConstants.TILE_SYNC:
                 return new SyncTile(this);
             default:
