--- Settings/src/com/android/settings/cyanogenmod/qs/QSTileHolder.java
+++ Settings/src/com/android/settings/cyanogenmod/qs/QSTileHolder.java
@@ -156,6 +156,10 @@
                 resourceName = "ic_qs_ambientdisplay_on";
                 stringId = R.string.qs_tile_ambient_display;
                 break;
+            case QSConstants.TILE_SCREENSHOT:
+                resourceName = "ic_qs_screenshot";
+                stringId = R.string.qs_screenshot_tile;
+                break;
             case QSConstants.TILE_SYNC:
                 resourceName = "ic_qs_sync_on";
                 stringId = R.string.qs_tile_sync;
